src/Congo.WebApi/Data/ProductAccess/InsertProductHelper.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Congo.WebApi.Data.Models;
+using MediatR;
+
+namespace Congo.WebApi.Data.ProductAccess
+{
+    public class InsertProductHelper : IRequestHandler<InsertProductCommand, Product>
+    {
+
+        private readonly CongoContext _dbContext;
+
+        public InsertProductHelper(CongoContext dbContext) => _dbContext = dbContext;
+
+        public async Task<Product> Handle(InsertProductCommand request, CancellationToken cancellationToken)
+        {
+            var product = new Product
+            {
+                Name = request.Name,
+                Description = request.Description,
+                ImageUrl = request.ImageUrl,
+                Price = request.Price
+            };
+
+            _dbContext.Products.Add(product);
+
+            await _dbContext.SaveChangesAsync();
+
+            return product;
```

Comment:
A command does not return anything, it's a write operation, meanwhile a Query is a read

Reply:
Yes and no :)

The distinction between it returning or not is less important than "commands WRITE" and "queries READ". Returning the product ID from a write operation does not include any reads, so thats fine to do.

At work, we don't use MediatR but have our own inhouse command dispatcher, and we have three request types: Command, CommandWithResponse and Query. Since MediatR itself doesn't diffrentiate between Query and Command, you do it yourself with the names.